### PR TITLE
Only update top level docs for latest version

### PR DESCRIPTION
**Pull request by chrismcg**
Closes hexpm/hex_web#382

### CODE REVIEW

File: web/controllers/api/docs_controller.ex [hunk 1]
```diff
@@ -96,18 +96,27 @@ defmodule HexWeb.API.DocsController do
     pre_release     = release.version.pre != []
     first_release   = package.docs_updated_at == nil
 
+    latest_release = from(r in Release.all(package), where: r.has_docs == true or r.version == ^version)
+                     |> HexWeb.Repo.all
+                     |> Enum.reject(fn(r) -> r.version.pre != [] end)
+                     |> Release.sort
+                     |> List.first
```
Comment (ericmj) [inline web/controllers/api/docs_controller.ex]: Could we fetch just the version field here?

File: test/controllers/api/docs_controller_test.exs [hunk 2]
```diff
@@ -57,6 +57,21 @@ defmodule HexWeb.API.DocsControllerTest do
     conn = get build_conn(), "docs/sitemap.xml"
     assert conn.status == 200
     assert conn.resp_body =~ "https://hexdocs.pm/phoenix"
+
+    body = create_tarball([{'index.html', "HEYA"}])
+    conn = build_conn()
+    |> put_req_header("content-type", "application/octet-stream")
+    |> put_req_header("authorization", key_for("eric"))
+    |> post("api/packages/phoenix/releases/0.0.1/docs", body)
```
Comment (ericmj) [inline test/controllers/api/docs_controller_test.exs]: Indent the pipelines.

**Comment by ericmj** [comment]
Looks great. Just two comments then we are good to merge.

**Comment by chrismcg** [comment]
@ericmj have updated, is that what you meant by just fetching the version field?

**Comment by ericmj** [comment]
Perfect. Thank you!

**Pull request merged**